src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -725,6 +749,31 @@ def _add_tensorboard_logger_if_available(self):
             TensorBoardLogger(writer=tensorboard_callback.tb_writer)
         )
 
+    def _get_fake_dataloader(
+        self,
+        num_samples: int,
+        tokenizer: "PreTrainedTokenizerBase",  # noqa: F821
+    ):
+
+        # Rearrange inputs' keys to match those defined by model foward func, which
+        # seem to define how the order of inputs is determined in the exported model
+        forward_args_spec = inspect.getfullargspec(self.model.__class__.forward)
```

Comment:
how confident are we that this is a stable thing to depend on?

Reply:
I have a not so great reason, The same idiom, has been used before in our export flows to generate a fake sample input to traverse the graph and stood the test of time

Reply:
it is a standard library, should be fine